data/block.go

```diff
@@ -0,0 +1,95 @@
+package data
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core"
+	nodeData "github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-core-go/data/receipt"
+	"github.com/multiversx/mx-chain-core-go/data/rewardTx"
+	"github.com/multiversx/mx-chain-core-go/data/smartContractResult"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+)
+
+// SaveBlockData holds the filtered block data that will be received on push events
+type SaveBlockData struct {
+	Hash      string                                              `json:"hash"`
+	Txs       map[string]*transaction.Transaction                 `json:"txs"`
+	Scrs      map[string]*smartContractResult.SmartContractResult `json:"scrs"`
+	LogEvents []Event                                             `json:"events"`
+}
+
+// InterceptorBlockData holds the block data needed for processing
+type InterceptorBlockData struct {
+	Hash          string
+	Body          nodeData.BodyHandler
+	Header        nodeData.HeaderHandler
+	Txs           map[string]*transaction.Transaction
+	TxsWithOrder  map[string]TransactionWrapped
```

Comment:
why missing pointer? Here and in all occurrences. This will induce shallow data copy when manipulated.

Reply:
done